packages/alfa-style/src/property/outline.ts

```diff
@@ -0,0 +1,209 @@
+          switch (value.type) {
+            case "keyword":
+              switch (value.value) {
+                case "thin":
+                  return Length.of(1, "px");
+
+                case "medium":
+                  return Length.of(3, "px");
+
+                case "thick":
+                  return Length.of(5, "px");
+              }
+
+            case "length":
+              return Resolver.length(value, style);
+          }
+        });
+      })
+  );
+
+  export namespace Width {
+    export type Specified =
+      | Length
+      | Keyword<"thin">
+      | Keyword<"medium">
+      | Keyword<"thick">;
+
+    export type Computed = Length<"px">;
+  }
+
+  /**
+   * @see https://drafts.csswg.org/css-ui/#outline-style
+   */
+  export const Style: Property<
+    Style.Specified,
+    Style.Computed
+  > = Property.of(
+    Keyword.of("none"),
+    Keyword.parse(
+      "auto",
+      "none",
+      "hidden",
+      "dotted",
+      "dashed",
+      "solid",
+      "double",
+      "groove",
+      "ridge",
+      "inset",
+      "outset"
+    ),
+    (style) => style.specified("outline-style")
+  );
+
+  export namespace Style {
+    export type Specified =
+      | Keyword<"auto">
+      | Keyword<"none">
+      | Keyword<"hidden">
+      | Keyword<"dotted">
+      | Keyword<"dashed">
+      | Keyword<"solid">
+      | Keyword<"double">
+      | Keyword<"groove">
+      | Keyword<"ridge">
+      | Keyword<"inset">
+      | Keyword<"outset">;
+
+    export type Computed = Specified;
+  }
+
+  /**
+   * @see https://drafts.csswg.org/css-ui/#outline-color
+   */
+  export const Color: Property<Color.Specified, Color.Computed> = Property.of(
+    Keyword.of("invert"),
+    css.Color.parse,
+    (style) =>
+      style.specified("outline-color").map((color) => {
+        if (color.type === "keyword" && color.value === "invert") {
+          return color;
+        }
+
+        return Resolver.color(color);
+      })
+  );
+
+  export namespace Color {
+    export type Specified = css.Color | Keyword<"invert">;
+
+    export type Computed =
+      | RGB<Percentage, Percentage>
+      | Current
+      | System
+      | Keyword<"invert">;
+  }
+
+  /**
+   * @see https://drafts.csswg.org/css-ui/#outline-offset
+   */
+  export const Offset: Property<
+    Offset.Specified,
+    Offset.Computed
+  > = Property.of(Length.of(0, "px"), Length.parse, (style) =>
+    style
+      .specified("outline-offset")
+      .map((offset) => Resolver.length(offset, style))
+  );
+
+  export namespace Offset {
+    export type Specified = Length;
+
+    export type Computed = Length<"px">;
+  }
+
+  /**
+   * @see https://drafts.csswg.org/css-ui/#outline
+   */
+  export const Shorthand = Property.Shorthand.of(
+    ["outline-width", "outline-style", "outline-color"],
+    (input) => {
+      let width: Width.Specified | undefined;
+      let style: Style.Specified | undefined;
+      let color: Color.Specified | undefined;
+
+      while (true) {
```

Comment:
We do seem to have a lot of these `while (true)` with `continue`s and `break` in style parsing. This is obviously due to the `||` operator in CSS which is order-independent.
I feel like we might need an helper to handle that to avoid duplicating the same pattern in all shorthand properties. Maybe something like `foo([Width.parse, Style.parse, Color.parse]) which would then return `[remaining, [width, style, color]])`? Typing might be annoying to due variadicity 🤔 

Anyway, not the topic for this PR…